private/tst/avb_streamhandler/src/IasTestSystemdWatchdog.cpp

```diff
@@ -0,0 +1,230 @@
+  ASSERT_NE(wdInterface, nullptr);
+  EXPECT_EQ(mWdManager->mWatchdogInterfaces.size(), 1);
+
+  wdInterface = nullptr;
+  wdInterface = mWdManager->createWatchdog(30000, "Odin");
+  ASSERT_NE(wdInterface, nullptr);
+  EXPECT_EQ(mWdManager->mWatchdogInterfaces.size(), 2);
+}
+
+TEST_F(IasTestSystemdWatchdog, destroyWatchdog)
+{
+  IasSystemdWatchdogManager *altWdManager = nullptr;
+  altWdManager = new (std::nothrow) IasSystemdWatchdogManager(mDltContext);
+  IasWatchdogInterface *wdIf_1, *wdIf_2;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+  wdIf_1 = wdIf_2 = nullptr;
+
+  wdIf_1 = mWdManager->createWatchdog(30000, "Milky Way");
+  ASSERT_NE(wdIf_1, nullptr);
+
+  wdIf_2 = altWdManager->createWatchdog(30000, "Andromeda");
+  ASSERT_NE(wdIf_2, nullptr);
+
+  res = mWdManager->destroyWatchdog(nullptr);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cParameterInvalid);
+
+  res = mWdManager->destroyWatchdog(wdIf_2);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cObjectNotFound);
+
+  res = mWdManager->destroyWatchdog(wdIf_1);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = altWdManager->destroyWatchdog(wdIf_2);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+}
+
+TEST_F(IasTestSystemdWatchdog, removeWatchdog)
+{
+  IasWatchdogInterface *wdIf = nullptr;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+
+  wdIf = mWdManager->createWatchdog(30000, "Thor");
+  ASSERT_NE(wdIf, nullptr);
+
+  res = mWdManager->removeWatchdog(wdIf);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = mWdManager->removeWatchdog(wdIf);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cObjectNotFound);
+
+  res = mWdManager->removeWatchdog(nullptr);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cParameterInvalid);
+}
+
+TEST_F(IasTestSystemdWatchdog, getCurrentRawTime)
+{
+  uint64_t curr_timestamp = 0;
+
+  curr_timestamp = mWdManager->getCurrentRawTime();
+  EXPECT_NE(curr_timestamp, 0);
+}
+
+TEST_F(IasTestSystemdWatchdog, registerWatchdog)
+{
+  IasWatchdogInterface *wdIf = nullptr;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+
+  wdIf = mWdManager->createWatchdog();
+  ASSERT_NE(wdIf, nullptr);
+  res = wdIf->registerWatchdog();
+  EXPECT_EQ(res, IasWatchdogResult::cWatchdogNotPreconfigured);
+
+  wdIf->setTimeout(30000);
+  wdIf->setName("Eitri");
+  res = wdIf->registerWatchdog();
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = wdIf->unregisterWatchdog();
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = wdIf->registerWatchdog(30000, "");
+  EXPECT_EQ(res, IasWatchdogResult::cParameterInvalid);
+
+  res = wdIf->registerWatchdog(0, "Sif");
+  EXPECT_EQ(res, IasWatchdogResult::cParameterInvalid);
+
+  res = wdIf->registerWatchdog(30000, "Sif");
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = wdIf->registerWatchdog(30000, "Sif");
+  ASSERT_EQ(res, IasWatchdogResult::cAlreadyRegistered);
```

Comment:
Nitpick: `EXPECT_EQ` here?

Reply:
Well, this was the end of the test, so I was like it won't matter...

Reply:
True that, but if someday more tests are added below... is one less thing to worry about.

Reply:
Agreed. Modified in the next rev.

Reply:
Done